### PR TITLE
Add Samba4 formula

### PR DESCRIPTION
**Pull request by bySabi**

### CODE REVIEW

**Comment by mikemcquaid** [comment]
This should be either an update for the `samba` formula or submitted to `homebrew-versions.

File: Library/Formula/samba4.rb [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+  conflicts_with 'samba', :because => 'don´t install both packages'
+  
+  depends_on "pkg-config" => :build
+  depends_on "gnutls" => :recommended
+  depends_on "popt" => :recommended  
+  
+  skip_clean 'private'
+  skip_clean 'var/locks'
+
+  # Fixes
```
Comment (mikemcquaid) [inline Library/Formula/samba4.rb]: Fixes what? Can you please submit this patch to the upstream developers of this project and add a link to the upstream patch submission in a comment in the formula file. Thanks!

**Comment by jacknagel** [comment]
Please rework this as an update to the existing samba formula, thanks.

File: Library/Formula/samba4.rb [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+  patch :DATA
+
+  def install
+    system "./buildtools/bin/waf configure", "--prefix=#{prefix}",
```
Comment (mistydemeo) [inline Library/Formula/samba4.rb]: `waf` and its argument need to be broken up into separate arguments, e.g.

```ruby
system "./buildtools/bin/waf configure", "--prefix=#{prefix}",
# ...
```

The same is true for subsequent lines.

Comment (bySabi) [inline Library/Formula/samba4.rb]: I guest some hours ago when formula don´t build :-)
still working on this. Stuck on [bug#11056](https://bugzilla.samba.org/show_bug.cgi?id=11056)

Comment (mistydemeo) [inline Library/Formula/samba4.rb]: All right, closing for now - but if the install name bug gets fixed, please resubmit!

Comment (bySabi) [inline Library/Formula/samba4.rb]: The install name bug could take sometime to be fixed by samba team. Is posible we can add like "samba4" formula on "homebrew-versions" setting build dir folder same has install dir, ex: /usr/local/Cellar/samba/4.1.16, and disable clean build dir after brew install?

**Pull request closed without merging**